### PR TITLE
Support nacos discovery

### PR DESCRIPTION
**Pull request by wForget**
### What is the purpose of the change

Support nacos discovery.

Make dist package:

```
mvn clean install -DskipTests -Ddiscovery=nacos
```

To configure during installation (deploy-config/linkis-env.sh):

```
DISCOVERY=NACOS
NACOS_SERVER_ADDR=127.0.0.1:8848
```

Nacos server screenshot:

image

Verify: successfully execute the shell script through the Linkis UJESClient.

image

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Comment by peacewong** [comment]
ping @jackxu2011 @jacktao007

**Comment by jackxu2011** [comment]
i see the nacos support feature is moved to 1.5.0, and spring cloud will upgrade to 2.7.x. then the config alse can be simplify?

File: linkis-dist/bin/install.sh [hunk 1]
```diff
@@ -320,34 +328,48 @@ fi
...
+  sed -i ${txt}  "s#linkis.app.version:.*#linkis.app.version: $LINKIS_VERSION-$currentTime#g" $LINKIS_HOME/conf/application-eureka.yml
+
+  ##server application.yml
+  sed -i ${txt}  "s#defaultZone:.*#defaultZone: $EUREKA_URL#g" $LINKIS_HOME/conf/application-linkis.yml
+
+  sed -i ${txt}  "s#defaultZone:.*#defaultZone: $EUREKA_URL#g" $LINKIS_HOME/conf/application-engineconn.yml
 
-  sed -i ${txt}  "s/# prefer-ip-address:/prefer-ip-address:/g" $LINKIS_HOME/conf/application-linkis.yml
-  sed -i ${txt}  "s/# instance-id:/instance-id:/g" $LINKIS_HOME/conf/application-linkis.yml
+  if [ "$EUREKA_PREFER_IP" == "true" ]; then
```
[the remark this answers is not in the document]
Comment (wForget) [reply linkis-dist/bin/install.sh:346]: should not, it points to https://github.com/apache/linkis/blob/f4cea4796ffcee67bdc6bb845e9c40eb8530139a/linkis-dist/deploy-config/linkis-env.sh#L115

**Comment by wForget** [comment]
> i see the nacos support feature is moved to 1.5.0, and spring cloud will upgrade to 2.7.x. then the config alse can be simplify?

Do you have any suggestions for simplification?

File: linkis-dist/bin/install.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackxu2011) [reply linkis-dist/bin/install.sh:346]: > > i see the nacos support feature is moved to 1.5.0, and spring cloud will upgrade to 2.7.x. then the config alse can be simplify?
> 
> Do you have any suggestions for simplification?

if spring upgrade to 2.7.x, we can use   spring.config.import import the nacos or euraka config

**Comment by jackxu2011** [comment]
> > i see the nacos support feature is moved to 1.5.0, and spring cloud will upgrade to 2.7.x. then the config alse can be simplify?
> 
> Do you have any suggestions for simplification?

if spring upgrade to 2.7.x, we can use spring.config.import import the nacos or euraka config

File: linkis-dist/bin/install.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackxu2011) [reply linkis-dist/bin/install.sh:346]: I mean change the EUREKA_PREFER_IP to PREFER_IP_ADDRESS.  Now spring-clould-alibaba-nacos just support ip, but the nacos also can support hostname.

**Review by peacewong — COMMENTED** [review]
Need to add nacos compilation and deployment documents

**Comment by peacewong** [comment]
ping~ @casionone

**Comment by peacewong** [comment]
ping~ @wForget

**Comment by wForget** [comment]
@peacewong @jackxu2011 Thanks for your review, this PR is outdated, I will re-adapt master and send a new PR.

**Pull request closed without merging**